### PR TITLE
PLANNER-1352: Development Guide

### PR DESCRIPTION
**Pull request by yurloc**
* Initial getting started instructions done in 4fb1e14a214d1e3c0769f5931d198b98664f445f.
* Project structure improved and documented, Javadocs added in #44.
* This PR adds development guide for the backend.

### CODE REVIEW

File: optaweb-vehicle-routing-backend/pom.xml [hunk 1]
```diff
@@ -72,6 +72,12 @@
       <artifactId>graphhopper-reader-osm</artifactId>
       <version>0.11.0</version>
     </dependency>
+    <!-- Enable Spring Boot Automatic Restart, see Development Guide in Readme to learn how to use it. -->
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-devtools</artifactId>
```
Comment (ge0ffrey) [inline optaweb-vehicle-routing-backend/pom.xml]: Side question:
Does spring-boot-devtools work well for you?
I've seen reports on stackoverflow that drools-core.jar and spring-boot-devtools don't get along. Or was it with org.reflections?

File: optaweb-vehicle-routing-backend/README.adoc [hunk 2]
```diff
@@ -59,3 +73,39 @@ Services are injected into plugins, which moves the burden of their
 construction and dependency resolution to the IoC container.
 Secondly, plugins implement services' output boundary interfaces to handle
 use case results, e.g. persisting changes to database, sending response to web UI.
+
+== Development Guide
+
+=== Automatic Restart
+
+https://docs.spring.io/spring-boot/docs/2.1.1.RELEASE/reference/htmlsingle/#using-boot-devtools-restart[Automatic restart]
```
Comment (ge0ffrey) [inline optaweb-vehicle-routing-backend/README.adoc]: Any way we can replace that "2.1.1.RELEASE" with "latest" or "LATEST" or "latestFinal" and it still works?
Our docs have such a symlink, I would expect theirs to have that too.

File: optaweb-vehicle-routing-backend/pom.xml (hunk 1 above)
Comment (yurloc) [reply optaweb-vehicle-routing-backend/pom.xml]: For the reference, I have found these:
* https://stackoverflow.com/questions/52606152/optaplanner-got-java-reflection-error-in-spring-boot
* https://stackoverflow.com/questions/35746370/optaplanners-drools-working-memory-is-empty

I haven't noticed any issues so far. We'll see. Thanks for mentioning this.

File: optaweb-vehicle-routing-backend/README.adoc (hunk 2 above)
Comment (yurloc) [reply optaweb-vehicle-routing-backend/README.adoc]: Yes, they use `current`. I'll change it.

Ideal solution would be keeping the version synchronized with our Spring Boot dependency version and verifying the link doesn't break with version upgrade during CI build.

I have filed #49.

**Pull request merged**